docs/src/Box.doc.js

```diff
@@ -419,11 +419,54 @@ card(
       />
     </MainSection.Subsection>
     <MainSection.Subsection
-      title="Using role"
+      title="Using 'as' property"
       description={`
-    Setting the  \`role\` property on Box classifies the Box as the semantically appropriate HTML element. For example, setting \`role="banner"\` will designate that Box to be the equivalent of a \`<header>\` within the page, allowing assistive technology to classify the Box appropriately.
+      By default, the Box component renders a \`div\` element, which doesn't provide much meaning to the user or assistive technology. Use the \`as\` prop to inform which HTML component should be rendered by the Box component to ensure a meaningful experience.
```

Comment:
we could add

"a \`div\`, non-semantic element or an element that  communicates nothing about its content, which doesn't provide much meaning to the user or assistive technology."

"Use the \`as\` prop to replace  \`div\` with semantic HTML elements that define its content. Semantic elements convey their meaning both to the user, ensuring the same rich user experience via assistive technologies, and the browser."